Assets/Scripts/Utility/AssetInjection/TextureReplacement.cs

```diff
@@ -547,32 +547,63 @@ public static Material GetMobileBillboardMaterial(int archive, MeshFilter meshFi
                 string fileName = TextureFile.IndexToFileName(archive);
-                var textureFile = new TextureFile(Path.Combine(DaggerfallUnity.Instance.Arena2Path, fileName), FileUsage.UseMemory, true);
-
-                // Import all textures in this archive
-                importedTextures.Albedo = new Texture2D[textureFile.RecordCount][];
-                importedTextures.EmissionMaps = importedTextures.IsEmissive ? new Texture2D[textureFile.RecordCount][] : null;
-                for (int record = 0; record < textureFile.RecordCount; record++)
+                var textureFile = new TextureFile();                
+                if (textureFile.Load(Path.Combine(DaggerfallUnity.Instance.Arena2Path, fileName), FileUsage.UseMemory, true))
                 {
-                    int frames = textureFile.GetFrameCount(record);
-                    var frameTextures = new Texture2D[frames];
-                    var frameEmissionMaps = importedTextures.IsEmissive ? new Texture2D[frames] : null;
-
-                    for (int frame = 0; frame < frames; frame++)
+                    // Import all textures in this archive
+                    importedTextures.Albedo = new Texture2D[textureFile.RecordCount][];
+                    importedTextures.EmissionMaps = importedTextures.IsEmissive ? new Texture2D[textureFile.RecordCount][] : null;
+                    for (int record = 0; record < textureFile.RecordCount; record++)
                     {
-                        if (record != 0 || frame != 0)
-                            LoadFromCacheOrImport(archive, record, frame, importedTextures.IsEmissive, true, out tex, out emission);
+                        int frames = textureFile.GetFrameCount(record);
+                        var frameTextures = new Texture2D[frames];
+                        var frameEmissionMaps = importedTextures.IsEmissive ? new Texture2D[frames] : null;
+
+                        for (int frame = 0; frame < frames; frame++)
+                        {
+                            if (record != 0 || frame != 0)
+                                LoadFromCacheOrImport(archive, record, frame, importedTextures.IsEmissive, true, out tex, out emission);
+
+                            frameTextures[frame] = tex ?? ImageReader.GetTexture(fileName, record, frame, true);
+                            if (frameEmissionMaps != null)
+                                frameEmissionMaps[frame] = emission ?? frameTextures[frame];
+                        }
+
+                        importedTextures.Albedo[record] = frameTextures;
+                        if (importedTextures.EmissionMaps != null)
+                            importedTextures.EmissionMaps[record] = frameEmissionMaps;
+                    }
+                }
+                // Otherwise, check what files are available in the injected assets
+                else
+                {
+                    List<Texture2D[]> allAlbedo = new List<Texture2D[]>();
```

Comment:
Unlike the branch above (ie: the existing code), I don't use the materialReader's cache. I wasn't sure how to do it without knowing the number of frames in the record ahead of time.
If anyone has any suggestions/ideas, do share them

Reply:
@ajrb Mentioned it would be possible to use XML metadata for frame and record counts. That is a different approach however and I'd rather not delay this one any further at this stage.